packages/ui/src/components/wui-icon-box/index.ts

```diff
@@ -0,0 +1,68 @@
+import { html, LitElement } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+import { classMap } from 'lit/directives/class-map.js'
+import { globalStyles } from '../../utils/ThemeUtil'
+import '../wui-icon'
+import type { Size as IconSize } from '../wui-icon'
+import styles from './styles'
+
+type BackgroundSize = 'lg' | 'md' | 'sm'
+
+type Color =
+  | 'blue-100'
+  | 'error-100'
+  | 'fg-100'
+  | 'fg-200'
+  | 'fg-300'
+  | 'inverse-000'
+  | 'inverse-100'
+  | 'success-100'
+
+@customElement('wui-icon-box')
+export class WuiIconBox extends LitElement {
+  public static styles = [globalStyles, styles]
+
+  // -- state & properties ------------------------------------------- //
+  @property() public size: BackgroundSize = 'md'
+
+  @property() public backgroundColor: Color = 'blue-100'
+
+  @property() public iconColor: Color = 'blue-100'
+
+  // -- render ------------------------------------------------------- //
+  public render() {
+    let iconSize: IconSize = 'xxs'
+    switch (this.size) {
+      case 'lg':
+        iconSize = 'lg'
+        break
+      case 'md':
+        iconSize = 'xs'
+        break
+      default:
+        iconSize = 'xxs'
+    }
+
+    const backgroundClasses = {
+      [`wui-opacity-${this.size === 'lg' ? 'sm' : 'md'}`]: true,
+      [`wui-color-${this.backgroundColor}`]: true
+    }
+
+    const sizeClasses = {
+      [`wui-size-${this.size}`]: true
+    }
+
+    return html`
+      <div class="${classMap(sizeClasses)}">
+        <div class="${classMap(backgroundClasses)}"></div>
```

Comment:
Why do we need extra absolute element here? This component seems like it can be acheived via following composition

```
<div>
  <wui-icon></wui-icon>
</div>
```

Reply:
We need the extra element, because we have to set an `opacity` on the `background`. There is a way to make it work with `rgba()`, but then we need to change our color config.